### DISCUSSION TITLE
Any pre-push hook that takes longer than 30 seconds fails with "Connection to github.com closed by remote host".

### DISCUSSION THREAD

**Discussion by josheverett** [community/community#31675] [Other Feature Feedback, Questions, & Ideas]
Starting this morning, _some_ of our team were unable to push to our primary repo, with a `Connection to github.com closed by remote host` message in the console. The number of team members affected slowly increased throughout the day. 

Eventually we narrowed down the problem to the push failing when our pre-push hook took longer than 30 seconds to complete. 

Indeed, attempting a push to a github repo with the following pre-push hook will fail with `Connection to github.com closed by remote host`:

```bash
sleep 31
```

In our case, the culprit was a yarn script that we'll call `do-stuff`, called within pur pre-push hook via `yarn do-stuff`.

In order to work around this problem, I removed `yarn do-stuff` from our pre-push hook, and instead added the following scripts to our `package.json`:

```json
{
    "branch": "yarn do-stuff && git push -u origin $(git rev-parse --abbrev-ref HEAD)",
    "push": "yarn do-stuff && git push",
}
```

This works, but the core question remains unanswered: Why did this start happening today?

Is this 30 second limit enforced by GitHub? And if so, was this limit recently lowered from its previous value?

Note that changing SSH config settings on your local machine has no effect on this problem (despite some red herrings to be found on StackOverflow).

A similar discussion (also from today) [may be found here](https://github.com/orgs/community/discussions/31651) but does not offer any explanation about the underlying problem.

**Comment by Wytrykus** [comment]
I'm experiencing the same since this morning, trying to push to a repo with some husky pre-push hook which used to work all the time before.
I've changed my SSH settings as suggested here: https://github.com/zeam-vm/nx_evision/issues/6 - which didn't help at first.
(My temporary workaround was to deactivate the pre-push hook).

However, I got the "Connection to github.com closed by remote host" message much sooner, not only after 30 seconds. So I've set the `ServerAliveInterval` to 5 seconds, e.g. this is the relevant part of my ~/.ssh/config file:
```
Host github.com
    ServerAliveInterval 5
    ServerAliveCountMax 30
```
With these settings, so far it works again!

**Reply by BaseInfinity** [reply]
Can confirm, this solution works  seems like pinging github every 5 seconds keeps the connection alive

**Reply by ryanstevens** [reply]
@Wytrykus we just tried this solution and it works great!

**Reply by josheverett** [reply]
@Wytrykus I'm able to push again today without any config changes, so I'm not sure if the ssh config actually fixed anything. I added `sleep 120` to our pre-push check, and even then I'm still able to push. I suspect github rolled back whatever change caused this issue.

**Reply by Wytrykus** [reply]
Ok, great! 🙂
